MdePkg/Include/Base.h

```diff
@@ -885,8 +885,14 @@ STATIC_ASSERT (ALIGNOF (__VERIFY_UINT32_ENUM_SIZE) == sizeof (__VERIFY_UINT32_EN
 **/
 #define BASE_CR(Record, TYPE, Field)  ((TYPE *) ((CHAR8 *) (Record) - OFFSET_OF (TYPE, Field)))
 
-// FIXME: Upstream general variants of these macros.
-#define DEBUG_RAISE()     ASSERT (FALSE)
+#define DEBUG_RAISE()                                                                                   \
+    if ((PcdGet8 (PcdDebugRaisePropertyMask) & DEBUG_PROPERTY_DEBUG_ASSERT_ENABLED) != 0) {             \
+      ASSERT (FALSE);                                                                                   \
+    } else if ((PcdGet8 (PcdDebugRaisePropertyMask) & DEBUG_PROPERTY_DEBUG_PRINT_ENABLED) != 0) {       \
+      DEBUG ((DEBUG_ERROR, "SecurePE: PE format specification is violated\n"));                         \

```

Comment:
This should have a more generic error. Something like `"DEBUG RAISE: Constraint violation in %a:%a:%u\n", __FILE__, __func__, __LINE__`

Reply:
Also, change to DEBUG_WARN, in my opinion.